### ISSUE TITLE
Crash on WMV playback attempt

### ISSUE THREAD

**Issue by bjstarosta** [Librelancer/Librelancer#68]
Currently this exception is thrown on my machine (Win 10 LTSC 1809) when the intro movie is supposed to play.

```
HRESULT: [0x80004002], Module: [General], ApiCode: [E_NOINTERFACE/No such interface supported], Message: No such interface supported

   at SharpDX.Result.CheckError()
   at SharpDX.ComObject.QueryInterface(Guid guid, IntPtr& outPtr)
   at SharpDX.ComObject.QueryInterface[T]()
   at LibreLancer.Media.VideoPlayerWMF.PlayFile(String filename)
   at LibreLancer.Media.VideoPlayer.PlayFile(String filename)
   at LibreLancer.IntroMovie..ctor(FreelancerGame game, Int32 index)
   at LibreLancer.FreelancerGame.Load()
   at LibreLancer.Game.Run()
   at lancer.MainClass.<>c__DisplayClass0_0.<Main>b__0()
   at LibreLancer.AppHandler.Run(Action action, Action onCrash)
```

I've looked into this a little bit and the only thing I found was to do with Windows updates for Windows 7, which wasn't particularly helpful, and COM object debugging is a massive PITA. I'm not sure at the moment if this is Windows 10 specific, or my setup specific (missing software of some sort?), or just your usual bugs introduced by Microsoft. Either way, I have a [commit](https://github.com/bjstarosta/Librelancer/commit/8832b59fb379deb398199cc273b0fd5e99001a43) for a more graceful bypass of playback instead of outright crash in my forked repo, I can submit a pull request if you want.

Also, I've not tested it on my Linux install yet, but can Mono even handle WMV COM playback on Linux? If not, it might not be a bad idea to say migrate to [this](https://archive.codeplex.com/?p=libvlcnet) which is going to be crossplatform by design.

**Comment by CallumDev** [comment]
Hi.
I find this interesting. Usually it only fails to QueryInterface on N versions of windows, which I added a check for. I've got a Win 10 LTSC virtual machine that I can't seem to reproduce this on, so it would be helpful if you make a pull request with that commit, though I'd like it better if you can write an error to the log saying QueryInterface failed on Media Foundation.

On Linux it invokes system mpv as that can draw to anything with an OpenGL context. LibVlc is nice but given the patent situation I can't actually redistribute anything in this repo that plays WMV/WMA. The intro movies are a nice-to-have so just fixing it to print an error instead of crashing is preferred :)

**Comment by CallumDev** [comment]
Fixed by pull request #69
